api/ApiRouter.inc

```diff
@@ -73,7 +76,35 @@ class ApiRouter
         throw new InvalidAPI();
     }
 
-    public static function get_router()
+    public function request(bool $raw = false)
+    {
+        if ($raw) {
+            return file_get_contents('php://input');
+        } else {
+            $json_object = json_decode(file_get_contents('php://input'), true);
+            if ($json_object === null) {
+                throw new InvalidValue("Content was not valid JSON");
+            }
+            return $json_object;
+        }
+    }
+
+    public function response(bool $raw = false): string
```

Comment:
We wouldn't need this if it was incorporated as above.

Reply:
Because I wanted to keep the encoding and decoding together and they got moved in here, breaking them out into clear functions makes it clearer what's going on IMHO.